libraries/AP_Scripting/lua_scripts.cpp

```diff
@@ -37,6 +37,24 @@ lua_scripts::lua_scripts(const AP_Int32 &vm_steps, const AP_Int32 &heap_size, co
     _heap = hal.util->allocate_heap_memory(heap_size);
 }
 
+lua_scripts::~lua_scripts() {
+    repl_cleanup();
+
+    while (scripts != nullptr) {
```

Comment:
This should actually be the end of the lua_scrpts::run() function. Run already manages loading the scripts, and should be responsible for also unloading and closing all the state, rather then bouncing it out to the destructor. It shouldn't actually change the behavior from what you have, but it contains it a lot better logically.

Reply:
I have moved the lua state cleanup to the end of the run function and left the freeing of heap to the destuctor.